test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+    });
+  });
+
+  describe('isImagesInfoActive()', function() {
+    it('should return true', function() {
+      createDomNodes([
+        createImg()
+      ]);
+      window.NCC.imageChecker.showImagesInfo();
+
+      expect(window.NCC.imageChecker.isImagesInfoActive()).toEqual(true);
+    });
+
+    it('should return false', function() {
+      createDomNodes();
+      window.NCC.imageChecker.showImagesInfo();
+
+      expect(window.NCC.imageChecker.isImagesInfoActive()).toEqual(false);
+    });
+  });
+
+  describe('_getTruncatedImageUrl()', function() {
+    it('should truncate url that does not fit bounding box', function() {
+      expect(window.NCC.imageChecker._getTruncatedImageUrl({
+        width: 50,
+        url: '1234567890'
+      })).toEqual('123...90');
+    });
+
+    it('should not truncate url that does fit bounding box', function() {
+      expect(window.NCC.imageChecker._getTruncatedImageUrl({
+        width: 50,
+        url: '12345678'
+      })).toEqual('12345678');
+    });
+  });
+
+  describe('_getElementTopLeft()', function() {
+    it('should use x y if available', function() {
+      createDomNodes([
+        createImg()
+      ]);
+      document.body.style = 'margin: 4px;';
+
+      expect(window.NCC.imageChecker._getElementTopLeft(img)).toEqual({top: 4, left: 4});
+    });
+
+    it('should use parent offset as fallback', function() {
+      createDomNodes([
+        createBackgroundImg()
+      ]);
+      document.body.style = 'margin: 2px;';
+
+      expect(window.NCC.imageChecker._getElementTopLeft(backgroundImg)).toEqual({top: 2, left: 2});
+    });
+  });
+
+  describe('_getImageCoverage()', function() {
+    it('should calculate using devicePixelRatio', function() {
+      expect(window.devicePixelRatio).toEqual(1);
+      expect(window.NCC.imageChecker._getImageCoverage({
+        naturalSize: {
+          width: 200,
+          height: 200
+        },
+        width: 100,
+        height: 100
+      })).toEqual(400);
+
+      window.devicePixelRatio = 2;
+      expect(window.NCC.imageChecker._getImageCoverage({
+        naturalSize: {
+          width: 200,
+          height: 200
+        },
+        width: 100,
+        height: 100
+      })).toEqual(200);
+    });
+  });
+
+  describe('_getNaturalSize()', function() {
+    it('should use natural size if available', function() {
+      createDomNodes([
+        createImg()
+      ]);
+
+      expect(window.NCC.imageChecker._getNaturalSize(img)).toEqual({width: 100, height: 80});
+    });
+
+    it('should use image natural size as fallback', function() {
+      createDomNodes([
+        createBackgroundImg()
+      ]);
+
+      expect(window.NCC.imageChecker._getNaturalSize(backgroundImg)).toEqual({width: 100, height: 80});
+    });
+  });
+
+  describe('_getBackgroundColor', function() {
+    it('should return high coverage color', function() {
+      expect(window.NCC.imageChecker._getBackgroundColor(300)).toEqual('red');
```

Comment:
@tabanliviu INFO - I know this is private, but the scale we're using here seems pure madness.